app/controllers/posts_controller.rb

```diff
@@ -3,11 +3,43 @@ class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
     @page = params.fetch(:page, 0).to_i
-    @posts = Post.limit(3).offset(@page * 3)
+    @posts = @user.posts.limit(3).offset(@page * 3)
+  end
+
+  def new
+    @user = User.find(params[:user_id])
+    @post = @user.posts.build
   end
 
   def show
     @post = Post.find(params[:id])
     @user = @post.author
+    @likes_count = @post.likes_counter
+  end
+
+  def like
+    @post = Post.find(params[:id])
+    @like = Like.create(user: current_user, post: @post)
+
+    respond_to do |format|
+      format.json { render json: { likes_count: @post.likes.count } }
+    end
+  end
```

Comment:
[OPTIONAL]
- [ ] Kindly remove all actions related to the **Likes** into the `app/controllers/likes_controller.rb` file to make your project look more clean.